training/src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -95,5 +98,20 @@ public Book updateBook(@RequestBody Book book, @PathVariable Long id) throws Boo
 
         return bookRepository.save(book);
     }
+
+    @GetMapping("/byPublisherAndByGenderAndByYear")
```

Comment:
IMO, I would personally use a PostMapping instead of a GetMapping if you're going to work with a request body. However, you could perform a GET request and use a @*RequestParam* for each parameter instead of @*RequestBody* since you're not working with sensitive data :)

Reply:
Also, replace "Gender" in the path and method name with "Genre" 😄